### PR TITLE
fix(Core/Mail): Do not show expired mails in packets.

### LINKED ISSUE #5920: Amount Pending Mails from AH should be automatically deleted on arrival of the money [$35]
Original report: https://github.com/chromiecraft/chromiecraft/issues/436 by @j-hdt

Additional information from @necropola:
>The **Money Mail**  is delayed (I've tested it) by `MailDeliveryDelay` from `worldserver.conf` (default: 3600 seconds = 1 hour, ChromieCraft: 150 seconds) and the **Amount Pending Mail** arrives immediately on sale of an item. I've also verified that  the **Amount Pending Mail** has exactly the expiration time that the **Money Mail** is delayed for. 
>
>The **Amount Pending Mail** stays in your Mailbox for quite a while after it has expired, because the cleanup job which deletes or returns expired mails does only run **every 6 hours**. It would probably be better (less irritating), if the cleanup job would run more often or at least at a configurable (`worldserver.conf`) frequency instead of being hardcoded:
https://github.com/azerothcore/azerothcore-wotlk/blob/b2ab5dba79c180608323c5ca02b9f815b6d35833/src/server/game/World/World.cpp#L2246-L2250
> In fact it looks like Blizzard intended the **Amount Pending Mails** to be automatically deleted on arrival of the money: https://www.bluetracker.gg/wow/topic/us-en/1777876864-auction-house-sale-pending-notice/
>
> Also, the **Estimated Delivery Time**  is always `00:00` which probably isn't correct, too.
>
>image

#### WHAT CLIENT DO YOU PLAY ON?
- [x] enGB
- [ ] enUS
- [ ] other (specify)

##### FACTION
- [x] Alliance
- [ ] Horde

##### CONTENT PHASE:
- [x] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:
When an auction you inserted into the auction house is bought, you receive a in-game mail. One hour later you receive the gold reduced by the costs. The notification mail for the sell does not disappear.

##### EXPECTED BLIZZLIKE BEHAVIOUR:
when receiving the mail containing the gold, the notification mail should disappear from the inbox.

##### STEPS TO REPRODUCE THE PROBLEM:
 - Which commands to use? Which NPC to teleport to?
 - Do we need to have debug flags on Cmake?
 - Do we need to look at the console while the bug happens?
 - Other steps
 - Use the ingame commands to identify the unique GUID of an ore/herb/npc:    .npc info  /  .gobject near
-->

1. insert auction into auction house
2. buy item from a different account
3. see the first mail arrive
4. one hour later when the gold arrives, the first mail should disappear

##### EXTRA NOTES:
There might be different behavior if you are online during the sell or the 1h period - or open the mail.
but in general those notification mails could be flagged with a 1h disappearing timer instead of 30 days.

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/b9ae587b6b37355448c30301f7cccd50175117cd

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

<bountysource-plugin>

---
There is a **[$35 open bounty](https://www.bountysource.com/issues/98542579-amount-pending-mails-from-ah-should-be-automatically-deleted-on-arrival-of-the-money?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by UltraNix**
Fixed #5920

## Issues Addressed:

- Closes #5920
- Closes https://github.com/chromiecraft/chromiecraft/issues/436

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

insert auction into auction house
buy item from a different account
see the first mail arrive
one hour(can be modified in worldserver.conf) later when the gold arrives, the first mail should disappear

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
Is auctionhouse field deprecated now? If so drop the db column too

File: data/sql/updates/pending_db_world/rev_1644150005753898600.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1644150005753898600');
+
+ALTER TABLE `mail` DROP COLUMN `auctionId`;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1644150005753898600.sql:3]: wrong database, you have a world file, but should be in character database :P

**Review by Nyeriah — APPROVED** [review]

**Comment by Nyeriah** [comment]
Gotta check if the cache is updating correctly (.cache info $name), otherise seems good to me

**Comment by pklloveyou** [comment]
Tested

**Comment by temperrr** [comment]
Tested:

.cache info seymort 
Mail count 0

Sold wool cloth
.cache info seymort
mail count 2
waited 2 min (as updated in console)
Got gold, took it out
.cache info seymort
mail count 1 
(No mail in inbox)
Shows mail icon on minimap

So cache is not updating correctly then?

**Comment by UltraNix** [comment]
@temperrr Yes - cache is not cleared from that mail that is not displayed. It's gonna be deleted on server startup or after 6h. It is one in this way because of server performance (see `ReturnOrDeleteOldMails`).

**Pull request merged**